src/Functions/array/arrayRemove.cpp

```diff
@@ -0,0 +1,195 @@
+
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeNothing.h>
+#include <DataTypes/DataTypeNullable.h>
+
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/array/arrayRemove.h>
+
+#include <Interpreters/castColumn.h>
+
+namespace DB
+{
+
+namespace
+{
+
+ColumnPtr BuildAndExecuteFunction(
+    const std::string & func_name, const ColumnsWithTypeAndName & func_args, ContextPtr context, size_t expected_column_count)
+{
+    auto func = FunctionFactory::instance().get(func_name, context)->build(func_args);
+    return func->execute(func_args, func->getResultType(), expected_column_count, /* dry_run = */ false);
+}
+
+}
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+    extern const int LOGICAL_ERROR;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+DataTypePtr FunctionArrayRemove::getReturnTypeImpl(const DataTypes & arguments) const
+{
+    if (arguments.size() != 2)
+        throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH,
+            "Number of arguments for function {} doesn't match: passed {}, should be 2",
+            getName(), arguments.size());
+
+    const DataTypeArray * array_type = checkAndGetDataType<DataTypeArray>(arguments[0].get());
+    if (!array_type)
+        throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                        "First argument for function {} must be an array but it has type {}.",
+                        getName(), arguments[0]->getName());
+
+    return arguments[0];
+}
+
+ColumnPtr FunctionArrayRemove::executeImpl(
+    const ColumnsWithTypeAndName & arguments,
+    const DataTypePtr & return_type,
+    size_t input_rows_count) const
+{
+    const auto * return_type_array = checkAndGetDataType<DataTypeArray>(return_type.get());
+    if (!return_type_array)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Return type for function {} must be Array, got {}", getName(), return_type->getName());
+
+    if (typeid_cast<const DataTypeNothing *>(return_type_array->getNestedType().get()))
+        return return_type->createColumnConstWithDefaultValue(input_rows_count);
+
+    const auto & arr_arg_column = arguments[0].column->convertToFullColumnIfConst();
+    const auto * arr_col = checkAndGetColumn<ColumnArray>(arr_arg_column.get());
+    if (!arr_col)
+        throw Exception(ErrorCodes::ILLEGAL_COLUMN,
+            "First argument for function {} must be Array, got {}", getName(), arguments[0].column->getName());
+
+    const auto & arr_data_col = arr_col->getDataPtr();
+    const auto & arr_offsets = arr_col->getOffsets();
+    size_t arr_elements_count = arr_data_col->size();
+
+    const auto & arr_data_type = assert_cast<const DataTypeArray &>(*arguments[0].type).getNestedType();
+    const auto & elem_type = arguments[1].type;
+    bool elem_is_const_null = elem_type->onlyNull();
+
+    ColumnPtr filter_col;
+    if (elem_is_const_null)
```

Comment:
```suggestion

    ColumnPtr filter_col;
    if (elem_type->onlyNull())
```

Very minor suggestion